app/tests/spec/lib/fxa-client.js

```
@@ -669,6 +669,54 @@ function (chai, $, sinon, FxaClient, p, testHelpers, Session, FxaClientWrapper,
       });
     });
 
+    describe('completeAccountUnlock', function () {
+      it('resends an account unlock email', function () {
+        sinon.stub(realClient, 'accountUnlockVerifyCode', function () {
+          return p();
+        });
+
+        return client.completeAccountUnlock()
+          .then(function () {
+            assert.isTrue(realClient.accountUnlockVerifyCode.called);
+          });
+      });
+    });
+
+    describe('resendAccountUnlockEmail', function () {
+      it('resends an account unlock email', function () {
+        sinon.stub(realClient, 'accountUnlockResendCode', function () {
+          return p();
+        });
+
+        return client.sendAccountUnlockEmail()
+          .then(function () {
+            assert.isTrue(realClient.accountUnlockResendCode.called);
+          });
+      });
+
+      it('limits the number of attempts', function () {
+        sinon.stub(realClient, 'accountUnlockResendCode', function () {
+          return p();
+        });
+        var triesLeft = Constants.ACCOUNT_UNLOCK_RESEND_MAX_TRIES;
+
+        // exhaust all tries
+        var promises = [];
+        for (var i = 0; i < triesLeft; i++) {
+          promises.push(client.sendAccountUnlockEmail(email, relier));
+        }
+
+        return p.all(promises)
+          .then(function () {
+            return client.sendAccountUnlockEmail(email, relier);
```

Add a comment that this silently succeeds.